includes/safe-svg-settings.php

```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+ * Safe SVG plugin settings.
+ *
+ * @package safe-svg
+ */
+
+/**
+ * SVG settings class.
+ */
+class safe_svg_settings {
+
+	/**
+	 * Set up the class
+	 */
+	public function __construct() {
+		add_action( 'admin_init', [ $this, 'settings_init' ] );
+		add_filter( 'pre_update_option_safe_svg_upload_roles', [ $this, 'update_capability' ], 10, 2 );
+	}
+
+	/**
+	 * Custom option and settings
+	 */
+	public function settings_init() {
+		register_setting( 'media', 'safe_svg_upload_roles' );
+
+		add_settings_section(
+			'safe_svg_settings',
+			__( 'Safe SVG Settings', 'safe-svg' ),
+			[ $this, 'safe_svg_settings_callback' ],
+			'media'
+		);
+
+		add_settings_field(
+			'safe_svg_roles',
+			__( 'User Roles', 'safe-svg' ),
+			[ $this, 'safe_svg_roles_cb' ],
+			'media',
+			'safe_svg_settings'
+		);
+	}
+
+	/**
+	 * Settings section callback function.
+	 *
+	 * @param array $args The settings array, defining title, id, callback.
+	 */
+	public function safe_svg_settings_callback( $args ) {
+		?>
+		<p id="<?php echo esc_attr( $args['id'] ); ?>">
+			<?php esc_html_e( 'Select user roles who can upload SVG files.', 'safe-svg' ); ?>
+		</p>
+		<?php
+	}
+
+	/**
+	 * User role field callback function.
+	 */
+	public function safe_svg_roles_cb() {
+		$user_roles   = get_editable_roles();
```

Comment:
This will show all roles, even those that can't do anything with media currently. Feels like we should trim this list down to only roles that currently have media upload capabilities to begin with, otherwise I could see confusion when someone sees `Subscriber` on this list, even though they can't upload any types of media

Reply:
I've updated this in the latest push, it will be filtered to roles that have the upload_files permission. I've also added a filter for these roles.